File: gatewayapi.spring/src/main/java/com/example/gatewayapi/controller/ErrorController.java
package com.example.gatewayapi.controller;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;


@Controller
@RequestMapping(value = "/api/v1/error", produces = MediaType.APPLICATION_JSON_VALUE)
public class ErrorController {

    private static final Logger log = LoggerFactory.getLogger(ErrorController.class);

    @RequestMapping(value = "/401", method = RequestMethod.GET)
    public ResponseEntity e401() {
        log.error("Not authorized");
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
    }
}

File: services/gateway/gateway-api/src/main/kotlin/com/example/gateway/api/rest/controller/ErrorController.kt
package com.example.gateway.api.rest.controller

import com.example.common.converter.TimeConverter
import com.example.gateway.api.spec.model.ExceptionResponse
import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/api/v1/error"], produces = [MediaType.APPLICATION_JSON_VALUE])
class ErrorController {

    @RequestMapping(value = ["/401", "/401/"], method = [RequestMethod.GET])
    @ResponseStatus(HttpStatus.UNAUTHORIZED)
    fun error401(): ExceptionResponse {
       return ExceptionResponse.newBuilder()
               .setCode(ExceptionResponse.ExceptionCode.UNAUTHORIZED)
               .setMessage("Invalid api key")
               .setTimestamp(TimeConverter.getOffsetDateTimeNow().toString())
               .build()
    }
}